internal/client/job.go

```diff
@@ -94,6 +101,74 @@ func (c *Project) setupLocalJobSystem(ctx context.Context) (isLocal bool, newCtx
 		// Inject the metadata about the client, such as the runner id if it is running
 		// a local runner.
 		newCtx = grpcmetadata.AddRunner(ctx, c.activeRunner.Id())
+	} else {
+		// We're about to run a remote op. We should check if we have a dirty local vcs,
+		// because the user may expect their local changes to be reflected in the remote
+		// op execution, and they won't.
+		gitDirtyErr := func() error {
```

Comment:
Tingling of whether this should just be extracted into a standalone unexported func with arguments.

Reply:
I have the same tingling - I tried it both ways. It's a very weird collection of arguments though, and only called here, so I went this route, but it feels a bit cute.

Reply:
Seems like something like this might be useful else where at some point too? I can see value it in being a func to call.